### DISCUSSION TITLE
Dropping .NET Framework Support

### DISCUSSION THREAD

**Discussion by SuRGeoNix** [SuRGeoNix/Flyleaf#264] [Announcements]
I'm planning to drop support of .NET Framework soon as Vortice will do the same at the next (3) version. 
If you think that it should still be supported please share your ideas!

**Comment by anguzo** [comment]
I am actually working on adding web camera support to Flyleaf - selecting camera, format(right now dshow plays in lowest quality) and playing it. Would be nice if it could be included in some release before dropping. 
Overall 3.6.11 has plenty of functionality right now, as embedded player it works very nice. Maybe making separate branch would make sense, because then people can fix some occasional bugs themselves. But I am not sure how Nuget packaging should work then.
I personally don't see reason why open-source project should contemplate to legacy support. Dropping support means possibly better performance and better coding experience and quality with all the modern syntactic sugar.

**Comment by SuRGeoNix** [comment]
@anguzo Any reason for using .NET Framework nowadays? I could understand if you have an old project that you want to integrate with FlyleafLib but creating a new one with .NET Framework I don't see the reason. Reference: https://learn.microsoft.com/en-us/dotnet/standard/choosing-core-framework-server

**Reply by anguzo** [reply]
In reality only legacy old projects, and even them can be most likely ported to .NET 6 using upgrade-assistant.
Though there is an interesting aspect with WPF. In .NET 6 they dropped support for XAML islands so for example an existing .NET Framework application using UWP UI elements will lose them in  .NET 6. And in WPF there is no good API for webcam. In our project we used UWP MediaCapture and CaptureElement. As I understand MediaCapture can still be used but then you have to sink to D3DImage layer for example. In Windows 11 there is also possibility to convert IP cameras to "webcam" which is visible to the Camera app. Actually even PTZ functionality worked if I remember correctly. But we couldn't embed it even in sample UWP project. Wrote to Microsoft, but doubt they will fix it, UWP is dead after all.
It's kinda funny but .NET Framework WPF with XAML islands can achieve more than .NET 6 WPF.
It might be viable while WinUI 3 is still in cradle sadly.
Though nowadays choosing Avalonia instead of WPF is considerable. Or Uno even.
The big thing would be ditching Media Foundation and other dependencies which are Windows bound. But then I am not sure how cross-platform webcam API and rendering would work. And also Vortice is heavily bound to Windows anyway.

Anyways, for my project CaptureElement is one of the limiting factors right now in question of transitioning to .NET 6. That's why I work on proper webcam support for Flyleaf, though there needs more work to be done before full transition.
